### ISSUE TITLE
Assembler / Import Bus slowdown

### ISSUE THREAD

**Issue by ekconcordian** [AppliedEnergistics/Applied-Energistics-2#3716]
Hey guys,

I am aware this question has been asked to death, and although I have spent the better part of my day looking up solutions to it, the only answers I see are either chalked up to "user error" or people encountering the issue in 1.7.10.

For the record, this is an SMP server with 5-15 active players at a time, running ATM3 Remix 0.0.2 (appliedenergistics2-rv5-stable-11 on 1.12.2), which runs forge-14.23.4.2759. I am in communication with the server owner trying to determine the source of the problem. We have determined that my setup is correct and should technically function with no lag, considering it is very small scale. It is also verified that this occurs while the server is running with max TPS and virtually no lag. 

Just to clarify, this is an ME Controller hooked up with a dense energy cell, and a leadstone fluxduct for a power supply. I have confirmed while testing this that it is supplied with power. There are open channels leading to the problem devices, which is pretty much every single import bus and assembler. I do not use any export buses currently, and the interfaces are importing instantly. Each assembler and import bus has been tested with no speed upgrades, maximum speed upgrades, and only a couple speed upgrades, but the issue still occurs. 

**The issue is, as follows:** during crafting operations that vary in size from a simple operation to make 20 pulverized iron, to making a stack of dense ME cables, the system will hang at random points either because the import buses attached to the machines simply stop importing (or slow down to a near crawl, despite upgrades), or because the associated molecular assembler assigned to craft whatever is needed will hang at a random recipe and either move very slowly or sit at 0% indefinitely. 
![image](https://user-images.githubusercontent.com/9299352/45064570-9bbea600-b082-11e8-8a29-369078519265.png)
_Controller setup._

![image](https://user-images.githubusercontent.com/9299352/45064610-cf99cb80-b082-11e8-913c-d55e7bf26353.png)
_Assembler setup._

![image](https://user-images.githubusercontent.com/9299352/45064631-e7714f80-b082-11e8-80a8-e7b3c4aed05f.png)
_Primary location where import buses are used._

I'm aware that this is a fairly inefficient setup, but I know it works just fine and should be good for early-game autocrafting because I have done this several times before in my own server / worlds, with no issue. Those servers were typically much smaller.

There are some abnormalities with the theory that it may be server performance lacking. First off, as stated above, TPS rests at about 20 all the time, it is a very powerful system. Secondly, I have confirmed with another member of the server who also uses their autocrafting system in the same dimension, that they are not running into these same issues. That would lead me to believe that it was user error, however, the primary counter-argument to that is that I have used this same exact setup before to no issue, as well as that sometimes the system works flawlessly / very well, and runs into very little slowdowns, but other times it will hang on trying to craft a single item--with no other queued tasks, with the items sitting in the molecular assembler supplied with a free channel and power, and plenty of space left in the storage system--at 0% in the molecular assembler. For over 10 minutes, before I reboot the system and it starts working again, typically only for a few seconds. No changes are made between it working very well and hanging on random items doing the same task (crafting dense cables).

My question: is there some reason this may be happening, perhaps in the way the mod is configured? I have read in a few random, kinda sketchy sources that AE2 implements slowdowns for certain chunks to preserve server resources, which is what I thought, although my system should be very low-intensive and shouldn't be hanging at 0% for so long. I also read that this section:
![image](https://user-images.githubusercontent.com/9299352/45064873-3966a500-b084-11e8-8a77-6d215e4d472b.png)
may be related to the issue. This is the default configuration inside AppliedEnergistics2.cfg, and I am curious if the mod expects the server to be lower performance than it is and is throttling certain areas to prevent the mod from taking up too many resources. I know many people are using complex AE2 systems.

I'm looking for clarification regarding this issue so I can coordinate with the server owner and find out if there's something we can change in regards to this configuration (or if this is a known issue, or if it's something I can fix on my own).

Thanks!
Happy to provide more information / pictures if needed.

**Comment by mindforger** [comment]
first thing: good report, very detailed and you already nailed down almost every known user error.

Just 3 more questions

Can you show us your storage area

Did you make sure everything is chunkloaded

Do you have interdimensional connections like the quantum network bridge and made sure it is chunkloaded as well? 

And to exclude a rare channel issue, do you use enderio me conduits?

**Comment by ekconcordian** [comment]
Hi, thanks for the quick response!

I confirmed that everything was chunkloaded; at the current moment, I do not have any interdimensional connections either. I do not use EnderIO ME conduits. I'll get back to you tomorrow when I can get on again to show you my storage area, but it's just an ME drive filled with 6 4k drives and 4 16k drives, the 16k drives were added before and did not help, despite tons of free space. Crafting storage was, at the time, two sets of 3 4K crafting units and two co-processors each (now upgraded to three sets).

Just wanted to point out also that I tried a common fix, which was to break the ME Controller and place it back down, "rebooting" the ME network (I actually have 5 controllers hooked together in my current setup, but only needed to break one), and that helped considerably after I submitted this post, and experienced no slowdowns afterwards. Tomorrow I will be able to test that method again when there's higher traffic on the server though, and will report back. It may have worked so well because there wasn't as many people on (only 7 or 8 at the time I think).
